### PR TITLE
added country codes fixes

### PR DESCRIPTION
**Pull request by pamzey**
Added fix to the country codes  and cancel appointment

### CODE REVIEW

File: app/controllers/api/v1/send_sms_controller.rb [hunk 1]
```diff
@@ -78,11 +87,22 @@ def initialize_variables
...
                               dob: patient.birthdate,
                               gender: patient.gender,
-                              person_phone: patient.phone
+                              person_phone: patient.phone.gsub(/\s+/, '')
                             }
                           end
   end
 
+  def fetch_configuration_from_global_property(facility_id)
+    GlobalProperty.where("property", "#{facility_id}_%")
```
Comment (fchiyenda) [inline app/controllers/api/v1/send_sms_controller.rb]: Why are we not matching the exact property?

File: config/application.yml.example [hunk 2]
```diff
@@ -103,30 +103,35 @@ idsr_sms:
...
-  sms_api_key: ''
-  sms_reminder: 'true'
-  next_appointment_message: 'Okondewa  dziwani kuti mukuyera kukabayisa katemera'
-  next_appointment_reminder_period: 'Instant reminder'
-  sms_activation: 'true'
-  show_sms_popup: 'true'
+eir_sms_configurations:
+  development:
+    sms_gateway_url: https://gateway.seven.io/api/sms
+    sms_api_key: c73C9ac143c557296F3E928f7b29ffB6da32473d93fa873e87Fb5521E1D7eBEB
```
Comment (fchiyenda) [inline config/application.yml.example]: Please remove the key from the github repo are sms popup and sms_activation site_based?

**Review by fchiyenda — CHANGES_REQUESTED** [review]

File: app/controllers/api/v1/send_sms_controller.rb (hunk 1 above)
Comment (pamzey) [reply app/controllers/api/v1/send_sms_controller.rb]: I have changed this

File: config/application.yml.example (hunk 2 above)
Comment (pamzey) [reply config/application.yml.example]: I have removed this

File: config/application.yml.example [hunk 3]
```diff
@@ -103,30 +103,35 @@ idsr_sms:
...
-  sms_reminder: 'true'
-  next_appointment_message: 'Okondewa  dziwani kuti mukuyera kukabayisa katemera'
-  next_appointment_reminder_period: 'Instant reminder'
-  sms_activation: 'true'
-  show_sms_popup: 'true'
+eir_sms_configurations:
+  development:
+    sms_gateway_url: ''
+    sms_api_key: ''
+    sms_reminder: true
```
Comment (fchiyenda) [inline config/application.yml.example:112]: Being that we have put these in site based config let us remove them from this config

File: app/controllers/api/v1/send_sms_controller.rb [hunk 4]
```diff
@@ -102,45 +140,17 @@ def guardian_phone
     relationships = service.find_relationships(filters)
     relationships[0].relation.try(:person_attributes)
                     .try(:find_by, person_attribute_type_id: PersonAttributeType.find_by_name('Cell Phone Number').id)
-                    .try(:value)
+                    .try(:value).gsub(/\s+/, '')
   end
 
   def service
     PersonRelationshipService.new(Person.find(params[:person_id]))
```
Comment (fchiyenda) [inline app/controllers/api/v1/send_sms_controller.rb:157]: Let us require the person_id param because sometimes the frontend it not sending it

**Review by fchiyenda — CHANGES_REQUESTED** [review]
When setting up application.yml let us just setup the environment the application is running in

File: app/controllers/api/v1/send_sms_controller.rb (hunk 4 above)
Comment (pamzey) [reply app/controllers/api/v1/send_sms_controller.rb:157]: I have made the changes to the function

**Review by fchiyenda — APPROVED** [review]
Need to handle error when saving configurations

**Pull request merged**